linkerd/app/src/env.rs

```diff
@@ -248,26 +261,34 @@ const DEFAULT_INITIAL_CONNECTION_WINDOW_SIZE: u32 = 1048576; // 1MB ~ 16 streams
 const DEFAULT_SHUTDOWN_GRACE_PERIOD: Duration = Duration::from_secs(2 * 60);
 
 // This configuration limits the amount of time Linkerd retains cached clients &
-// connections.
+// connections for a given destination ip:port, as referenced by the application
+// client.
 //
 // After this timeout expires, the proxy will need to re-resolve destination
 // metadata. The outbound default of 5s matches Kubernetes' default DNS TTL. On
 // the outbound side, especially, we want to use a limited idle timeout since
 // stale clients/connections can have a severe memory impact, especially when
-// the application communicates with many endpoints or at high concurrency.
-//
-// On the inbound side, we want to be a bit more permissive so that periodic, as
-// the number of endpoints should generally be pretty constrained.
-const DEFAULT_INBOUND_ROUTER_MAX_IDLE_AGE: Duration = Duration::from_secs(20);
-const DEFAULT_OUTBOUND_ROUTER_MAX_IDLE_AGE: Duration = Duration::from_secs(5);
+// the application communicates with many destinations.
+const ENV_OUTBOUND_DISCOVERY_IDLE_TIMEOUT: &str = "LINKERD2_PROXY_OUTBOUND_DISCOVERY_IDLE_TIMEOUT";
+const DEFAULT_OUTBOUND_DISCOVERY_IDLE_TIMEOUT: Duration = Duration::from_secs(5);
+
+// On the inbound side, we may lookup per-port policy or per-service profile
```

Comment:
tiny nit: should this be
```suggestion
// On the inbound side, we may lookup per-port policy or per-service-profile
```

Reply:
No, we may lookup per-service serviceprofiles.